libs/designer/src/lib/ui/panel/configureTemplatePanel/configureWorkflowsPanel/usePanelTabs.tsx

```diff
@@ -2,30 +2,84 @@ import type { TemplateTabProps } from '@microsoft/designer-ui';
 import { selectWorkflowsTab } from './tabs/selectWorkflowsTab';
 import { customizeWorkflowsTab } from './tabs/customizeWorkflowsTab';
 import { useIntl } from 'react-intl';
-import { useDispatch } from 'react-redux';
-import type { AppDispatch } from '../../../../core/state/templates/store';
-
-export const useConfigureWorkflowPanelTabs = ({
-  onClosePanel,
-}: {
-  onClosePanel: () => void;
-}): TemplateTabProps[] => {
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppDispatch, RootState } from '../../../../core/state/templates/store';
+import { useFunctionalState } from '@react-hookz/web';
+import type { WorkflowTemplateData } from '../../../../core';
+import { updateAllWorkflowsData } from '../../../../core/state/templates/templateSlice';
+import {
+  getWorkflownameFromWorkflowId,
+  getWorkflowsWithDefinitions,
+  initializeWorkflowsData,
+} from '../../../../core/actions/bjsworkflow/configuretemplate';
+
+export const useConfigureWorkflowPanelTabs = (): TemplateTabProps[] => {
   const intl = useIntl();
   const dispatch = useDispatch<AppDispatch>();
+  const { workflowsInTemplate, workflowState } = useSelector((state: RootState) => ({
+    workflowsInTemplate: state.template.workflows,
+    workflowState: state.workflow,
+  }));
 
   const hasError = false; // Placeholder for actual error state
   const isSaving = false; // Placeholder for actual saving state
 
+  const [selectedWorkflowsList, setSelectedWorkflowsList] =
+    useFunctionalState<Record<string, Partial<WorkflowTemplateData>>>(workflowsInTemplate); //TODO: update the initial value to the workflows in the template
```

Comment:
You can remove the TODO... I have a loadCustomTemplate code which runs on wizard open and that should initialize the existing workflows in the store. Since we dont have teh api or know the data it returns, so that is just returning {}